test/functional/unit_bypass_validation.test.js

```diff
@@ -29,17 +29,18 @@ describe('bypass document validation', function () {
             ok: 1,
             cursor: {
               firstBatch: [{}],
-              id: 23,
+              id: 0,
```

Comment:
could you remind me why this needed to be updated (and why it was working before)?

Reply:
An exhausted cursor always returns id `0`, so the driver knows there is nothing left to get more and can do its cleanup. Part of this cleanup is ending the session that the cursors owns, but if there is no session this does not need to happen. This test previously was executing on a mocked wire version that had no session support, so the test would not be leaking a session. Once the mocked server was updated to a wire version that supported sessions, this cleanup code was not running and the client was getting closed while the cursor still expected more results from the mocked server.